pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package pluginwatcher
+
+import (
+	"net"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"google.golang.org/grpc"
+)
+
+// PluginInfo contains gRPC connection info
+type PluginInfo struct {
+	SockPath string
+}
+
+func WatchPlugins(path string, outChan chan PluginInfo) error {
+	watcher, err := fsnotify.NewWatcher()
```

Comment:
Thanks a lot for implementing this change very quickly!

Haven't looked through the whole PR, but looking at this part, I am curious whether kubelet can re-discover all of plugins after it restarts. Do we need to scan through the directory initially?

Reply:
Thanks, done. PTAL!